### PR TITLE
Add fitbit goals

### PR DESCRIPTION
**Pull request by whiteshadow-gr**

### CODE REVIEW

**Comment by whiteshadow-gr** [comment]
Sooo. There are 2 issues. 

First is a minor optimization issue. Except sleep goals, no other endpoint had a date, so I am adding one. That means though that 1 record will be saved no matter what to the hat. I probably can avoid that with storageParameters but don't think it worths the effort. 

Second issue is with empty goals. Now that I am adding the date in fat goal for example the JSON has only the date that I added so it fails the validation. I dunno how I can avoid that

**Comment by augustinas** [comment]
Re 1, sound good, just make sure that the goals are synchronised every 7 days.
Re 2, check for the empty goals object case and don't add the date in this scenario.

File: dataplug-fitbit/app/org/hatdex/dataplugFitbit/Module.scala [hunk 1]
```diff
@@ -62,6 +62,11 @@ class Module extends AbstractModule with ScalaModule with AkkaGuiceSupport {
     fitbitSleepInterface: FitbitSleepInterface,
     fitbitWeightInterface: FitbitWeightInterface,
     fitbitLifetimeStatsInterface: FitbitLifetimeStatsInterface,
+    FitbitSleepGoalsInterface: FitbitSleepGoalsInterface,
```
Comment (whiteshadow-gr) [inline dataplug-fitbit/app/org/hatdex/dataplugFitbit/Module.scala]: Oh well. Just saw those. Why is it capital? Same below. How does that compile and work.... Sometimes Scala magic is beyond magical and that's not good

Comment (augustinas) [reply dataplug-fitbit/app/org/hatdex/dataplugFitbit/Module.scala]: It's just a variable name the developer comes up with. If you came up with a name that starts with a capital, it's a valid variable name. Why should it not compile? Now if you tried to start it with digit or another symbol...

**Pull request merged**